### PR TITLE
Fix reading of files in the root of a zip

### PR DESCRIPTION
**Pull request by andres1gb**
This is:

```
- [x] a bugfix
- [ ] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary

### Why this change is needed?
Because the project fails to read Xlsx files created by some generators (such as Google Campaign Manager) which puts xml files in the root of the zip container.

### CODE REVIEW

**Comment by andres1gb** [comment]
It seems that Scrutinizer fails to start the test. Looks more like an initialization issue than a failure with the test itself.

`configure: error: Package requirements (openssl >= 1.0.2) were not met:

Requested 'openssl >= 1.0.2' but version of OpenSSL is 1.0.1f

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables OPENSSL_CFLAGS
and OPENSSL_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.`

**Comment by oleibman** [comment]
You are correct about the Scrutinizer problem. If it remains unfixed, it will not stop your change from being merged. However, would it be possible to add a unit test which would fail without your change, but which succeeds with your change?

**Comment by andres1gb** [comment]
> You are correct about the Scrutinizer problem. If it remains unfixed, it will not stop your change from being merged. However, would it be possible to add a unit test which would fail without your change, but which succeeds with your change?

I have CM360 generated Xlsx files that would fail, so I could replace business-related data (that I can't disclose) while keeping the ZipFile structure and write a test that simply read that file. Without the change it won't be able to read any cell.

File: src/PhpSpreadsheet/Reader/Xlsx.php [hunk 1]
```diff
@@ -368,6 +368,9 @@ private function getFromZipArchive(ZipArchive $archive, $fileName = '')
         if (strpos($fileName, '//') !== false) {
             $fileName = substr($fileName, strpos($fileName, '//') + 1);
         }
+        // Relative paths generated by dirname($filename) when $filename
+        // has no path (i.e.files in root of the zip archive)
+        $fileName = (string) preg_replace('/^\.\//', '$1', $fileName);
```
Comment (oleibman) [inline src/PhpSpreadsheet/Reader/Xlsx.php:373]: Sorry for not noticing this earlier. Your regular expression works, but only accidentally. The value in `$1` should never be set given the regular expression you're using. I think setting the replacement value to the null string is what you want to do.

**Review by oleibman — CHANGES_REQUESTED** [review]

File: src/PhpSpreadsheet/Reader/Xlsx.php (hunk 1 above)
Comment (andres1gb) [reply src/PhpSpreadsheet/Reader/Xlsx.php]: Absolutely. I simplified the former regex (which grouped apart the path), and left the replacement that happens to be always null. It's already fixed.

**Review by oleibman — APPROVED** [review]

**Comment by oleibman** [comment]
Thank you for your contribution.

**Pull request merged**